File: 2020/01/02.js
var fs = require('fs');
var data = [];

const fileData = fs.readFileSync('./input', {encoding:'utf8', flag:'r'}); 
const lines = fileData.split("\n");
for (const line of lines) {
    data.push(+line);
}

for (var i = 0; i < data.length; ++i) {
    const num1 = data[i];
    for (var j = i + 1; j < data.length; ++j) {
        const num2 = data[j];
        for (var k = j + 1; k < data.length; ++k) {
            const num3 = data[k];
            if (num1 + num2 + num3 === 2020) {
                console.log(num1 * num2 * num3);
            }
        }
    }
}

File: 2020/01/02.swift
import Foundation

func calculateAnswer(_ numbers: [Int]) -> Int {
    let totalNumbers = numbers.count
    
    for i in 0..<totalNumbers {
        let num1 = numbers[i]
        
        for j in i+1..<totalNumbers {
            let num2 = numbers[j]
            if num1 + num2 > 2020 {
                continue
            }
            
            for k in j+1..<totalNumbers {
                let num3 = numbers[k]
                if num1 + num2 + num3 == 2020 {
                    return num1 * num2 * num3
                }
            }
        }
    }
    return -1
}

let contents = try! String(contentsOfFile: "Input")
let numbers = contents.split(separator: "\n")
                      .compactMap{Int($0)}
let answer = calculateAnswer(numbers)

print(answer)
assert(answer == 73616634, "Answer is incorrect!")
